tsdb/db.go

```diff
@@ -0,0 +1,86 @@
+package tsdb
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/prometheus/common/log"
+)
+
+const (
+	// BlockDataSeparator sets a separator for block data value.
+	BlockDataSeparator = "|"
+	// ParameterSeparator sets a separator between parameters in a block.
+	ParameterSeparator = ","
+	// FileExtension file extension for db files.
+	FileExtension = ".json"
+)
+
+//HashList use case HashList for the HashTable
+type HashList struct {
+	URL        string `json:"url"`         // stores the URL
+	HashNumber string `json:"hash-number"` // stores the hash number of a particular URL and type
+	Type       string `json:"type"`        // type of action
+}
+
+func init() {
+	if _, err := parse("storage/000000"); err != nil {
+		log.Infof("creating in-memory chain2: storage/0000000\n")
+		if err := saveToHDD("storage/000000", []byte("[]")); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func parse(path string) (*string, error) {
+	res, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, errors.New("file not existed. create a new chain")
+	}
+	str := string(res)
+	return &str, nil
+}
+
+func checkAndCreatePath(path string) {
+	array := strings.Split(path, "/")
+	array = array[:len(array)-1]
+	path = strings.Join(array, "/")
+	_, err := os.Stat(path)
+	if err != nil {
+		e := os.MkdirAll(path, os.ModePerm)
+		if e != nil {
+			panic(e)
+		}
+	}
+}
+
+func saveToHDD(path string, data []byte) error {
+	checkAndCreatePath(path)
+	e := ioutil.WriteFile(path, data, 0755)
+	if e != nil {
+		return e
+	}
+	return nil
+}
+
+//SaveHashTableToHDD saves the hashtable to HDD
```

Comment:
```suggestion
// SaveHashTable saves the hashtable to the disk.
```

Reply:
I also think that calling this as **hash.*** is confusing. This should be something like an index or indextable.